src/Microsoft.AspNetCore.BuildTools.ApiCheck/BreakingChange.cs

```diff
@@ -1,21 +1,51 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using ApiCheck.Description;
-
 namespace ApiCheck
 {
     public class BreakingChange
     {
-        public BreakingChange(ApiElement oldItem, string context = null)
+        public BreakingChange(string typeId, string memberId, ChangeKind kind)
+        {
+            TypeId = typeId;
+            MemberId = memberId;
+            Kind = kind;
+        }
+
+        public string TypeId { get; }
+        public string MemberId { get; }
+        public ChangeKind Kind { get; }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, obj))
+            {
+                return true;
+            }
+
+            return obj.GetType() == GetType() && Equals((BreakingChange)obj);
+        }
+
+        private bool Equals(BreakingChange other)
         {
-            Context = context;
-            Item = oldItem;
+            return string.Equals(TypeId, other.TypeId) && string.Equals(MemberId, other.MemberId) && Kind == other.Kind;
         }
-        public string Context { get; }
 
-        public ApiElement Item { get; }
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = TypeId.GetHashCode();
```

Comment:
Perhaps use https://github.com/aspnet/Common/tree/dev/shared/Microsoft.Extensions.HashCodeCombiner.Sources instead so we don't have random/arbitrary hash code combiners scattered in teh codez.